### DISCUSSION TITLE
Cumulative min/max over dimension?

### DISCUSSION THREAD

**Discussion by naught101** [pydata/xarray#5613] [Q&A] [answered]
Pandas has a nice function to calculate the cumulative maximum: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.cummax.html

What is the easiest way to do something similar in xarray over a single dimension?

I think the only way I can think of is the stack all of the other dimensions into a pseudo dimension, the run a groupby, apply, convert to pandas, cummax, and then convert back. But this seems like it would be pretty slow, and the code is ugly. Is there a better way?

Also wondering if a generic `DataArray.cumulative` that takes a simple `func(a,b)` argument might be useful.

**Comment by max-sixty** [comment, marked as answer]
Does `.rolling(...).max()` with a window length of the whole dimension do what you're looking for?

(ref https://github.com/pydata/xarray/issues/5215)

**Reply by naught101** [reply]
doh... of course, it does. Thank you!
